### PR TITLE
Fixes #29507 - Add puma-status for reporting

### PR DESCRIPTION
**Pull request by ehelms**
Replaces https://github.com/theforeman/foreman/pull/7578 with additional commit to ensure puma is 4.3.5 or higher.

### CODE REVIEW

**Comment by ehelms** [comment]
@tbrisker this would be nice to have in 2.1 for helping with debug

**Comment by tbrisker** [comment]
tests are failing due to gemfile conflicts. Also, I assume this needs SELinux changes and packaging as well?

**Comment by ehelms** [comment]
Packaging PR -- https://github.com/theforeman/foreman-packaging/pull/5031

I don't know if this requires SELinux changes, no issues were reported when testing. @lzap do you forsee any being needed?

**Comment by lzap** [comment]
The only problem I can see is that the socket will have `tmp_t` (or similar) file label. If there's anything confined that needs to connect to it, it will require a rule to connect/read/write to `tmp_t`. Ideally this should have been a separate label.

But I believe `pumactl` runs unconfined (command line utility), so no issues there.

**Comment by sthirugn** [comment]
Looks good, thank you for this PR.

**Comment by ehelms** [comment]
Turns out I misread the puma repository structure and the 4.3.5 branch is missing the commit we need: 

https://github.com/puma/puma/commits/4.3.5

**Comment by ares** [comment]
@ehelms this needs a rebase now, could you also please share an update about puma? There was some release already, so is this still blocked?

**Comment by ehelms** [comment]
We never got a Puma 4 release with the needed security fix we implemented thus this has sat stagnant. A move to Puma 5 will enable this. Feel free to leave it open with that caveat or close it to be re-investigated at a later date.

**Comment by ekohl** [comment]
Since https://projects.theforeman.org/issues/30490 `/run/foreman` should no longer be world readable so the security concern may no longer be valid.

**Comment by ezr-ondrej** [comment]
@ehelms can we close this as Ewoud mentioned? I believe he's right and the concern is no longer valid.

**Comment by ekohl** [comment]
> @ehelms can we close this as Ewoud mentioned? I believe he's right and the concern is no longer valid.

This is not what I meant this morning. I think that the blocker is no longer valid which means we can implement this. The feature itself is still valid.

**Comment by ehelms** [comment]
Yep, this is valid and will be do-able with Puma 5 which we are actively working on.

**Comment by ekohl** [comment]
> Yep, this is valid and will be do-able with Puma 5 which we are actively working on.

My argument was that the socket will live in `/run/foreman`. Currently that is mode 0700 so the security concern is no longer and issue so I don't see why we can't do this with Puma 4. With Puma 5 we can even better guarantee the security though.

**Comment by ekohl** [comment]
ping @ehelms

**Comment by ekohl** [comment]
https://github.com/theforeman/foreman/pull/8171 updates to Puma 5.1 which includes the fixes needed.

**Comment by ehelms** [comment]
Intending to pick this up next sprint when I work on updating to Puma 5.1 across the board.

**Comment by ezr-ondrej** [comment]
What is the plan for this PR? :)

**Comment by ehelms** [comment]
Rebased and cleaned this up. Also opened  packaging PR to consume this https://github.com/theforeman/foreman-packaging/pull/6322

**Comment by ekohl** [comment]
I think you meant to link https://github.com/theforeman/foreman-packaging/pull/6322. Is there also a related PR for Debian to maintain parity?

**Comment by ehelms** [comment]
Deb link https://github.com/theforeman/foreman-packaging/pull/6323

File: config/puma/production.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
+run_dir = Rails.root.join('tmp')
+
+# Store server info state.
+state_path File.join(run_dir, 'puma.state')
+state_permission 0640
```
Comment (ekohl) [inline config/puma/production.rb]: Looks like Rubocop is unhappy here

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by ekohl** [comment]
Looks like @ares wrote a tutorial for this: https://community.theforeman.org/t/puma-status-monitoring-not-working-at-the-moment/23154

**Comment by ehelms** [comment]
I updated the Rubocop issue and rebased. Also updates the packaging PRs:

 * https://github.com/theforeman/foreman-packaging/pull/6322
 * https://github.com/theforeman/foreman-packaging/pull/6323

**Review by ekohl — APPROVED** [review]
This change looks correct. Probably good to get packaging changes approved too before merging.

**Comment by evgeni** [comment]
RPM changes are good, Debian I need to look at, but I see no hurdles.

**Pull request merged**